video_store.py

```diff
@@ -0,0 +1,153 @@
+        self.url = URL
+        self.selected_video = selected_video
+        self.selected_customer = selected_customer
+
+    def check_out_video(self, customer_id, video_id):
+        query_params={
+            "customer_id": customer_id,
+            "video_id": video_id
+        }
+        print(query_params)
+        print(self.url+"/rentals/check-out")
+        response = requests.post(self.url+"/rentals/check-out", json = query_params)
+        
+        return response
+
+    def check_in_video(self, customer_id, video_id):
+        query_params={
+            "customer_id": customer_id,
+            "video_id": video_id
+        }
+        response = requests.post(self.url+"/rentals/check-in", json= query_params)
+        return response
+
+    def print_selected(self):
+        pass
+        if self.selected_video:
+            print(f"Video with id {self.selected_video} is selected.") 
+        else:
+            print("Video not found")
+
+    def create_video(self, title= "Default title", release_date= "Default release date", total_inventory= "default stock"):
+
+        query_params = {
+            "title": title,
+            "release_date": release_date,
+            "total_inventory": total_inventory,
+        }
+        response = requests.post(self.url+"/videos",json=query_params)
+        return response.json()
+
+    def list_videos(self): 
+        response = requests.get(self.url+"/videos")
+        return response.json()
+
+    def get_video(self, title=None, video_id=None, release_date= None):
+        
+        for video in self.list_videos():
+            if title:
+                if video["title"]==title:
+                    video_id = video["id"]
+                    self.selected_video = video
+            elif int(video_id) == video["id"]:
+                self.selected_video = video
+
+        if self.selected_video == None:
+            return "Could not find video in stock"
+
+        response = requests.get(self.url+f"/videos/{video_id}") 
+        return response.json()
+
+    def update_video(self,title=None, video_id=None, release_date=None, total_inventory=None): 
+        if not title:
+            title = self.selected_video["title"]
+        if not video_id:
+            video_id = self.selected_video["id"]
+        if not release_date:
+            release_date = self.selected_video["release_date"]
+        if not total_inventory:
+            total_inventory = self.selected_video["total_inventory"]
+
+        query_params = {
+            "id": video_id,
+            "title": title,
+            "release_date": release_date,
+            "total_inventory": total_inventory}
+        
+        response = requests.put(
+            self.url+f"/videos/{self.selected_video['id']}",
+            json=query_params
+            )
+        print("response:", response)
+        self.selected_video = response.json()
+        return response.json()
```

Comment:
One opportunity for improvement for many people for this project is handling error codes from the API requests. In general, when writing code to call an API, it's appropriate to check whether the request was successful or it triggered some sort of error. This is where HTTP response codes come in handy! We can usually assume that a 2XX response code means there was no error and any other response code means there was some sort of error. It's very common to have an if...elif...else chain to handle different ranges of response codes. 

For example, if I try to edit a video with an invalid inventory or date, the API will respond with an error code but no response body. This results in a `JSONDecodeError` when line 89 is executed. This could be avoided by printing a different message when the response code is an error code (ie. in the range 300-500).

This applies to all the other API calls as well.